internal/acceptance/entitlements_test.go

```diff
@@ -95,3 +103,122 @@ func TestAccServicePrincipalEntitlementsResourceOnAws(t *testing.T) {
 		}`,
 	})
 }
+
+type entitlement struct {
+	name  string
+	value bool
+}
+
+func (e entitlement) String() string {
+	return fmt.Sprintf("%s = %t", e.name, e.value)
+}
+
+func entitlementsStepBuilder(t *testing.T, c **httpclient.ApiClient, groupName string) func(entitlements []entitlement) step {
+	return func(entitlements []entitlement) step {
+		entitlementsBuf := strings.Builder{}
+		for _, entitlement := range entitlements {
+			entitlementsBuf.WriteString(fmt.Sprintf("%s\n", entitlement.String()))
+		}
+		return step{
+			Template: fmt.Sprintf(`
+			data "databricks_group" "example" {
+				display_name = "%s"
+			}
+			
+			resource "databricks_entitlements" "entitlements_users" {
+				group_id              = data.databricks_group.example.id
+				%s
+			}
+		`, groupName, entitlementsBuf.String()),
+			Check: func(s *terraform.State) error {
+				groupId := s.RootModule().Resources["data.databricks_group.example"].Primary.ID
+				var res iam.Group
+				ctx := context.Background()
+				err := (*c).Do(ctx, "GET", fmt.Sprintf("/api/2.0/preview/scim/v2/Groups/%s?attributes=entitlements", groupId),
+					httpclient.WithResponseUnmarshal(&res))
+				assert.NoError(t, err)
+				receivedEntitlements := make([]string, 0, len(res.Entitlements))
+				for _, entitlement := range res.Entitlements {
+					receivedEntitlements = append(receivedEntitlements, entitlement.Value)
+				}
+				expectedEntitlements := make([]string, 0, len(entitlements))
+				for _, entitlement := range entitlements {
+					if entitlement.value {
+						expectedEntitlements = append(expectedEntitlements, strings.ReplaceAll(entitlement.name, "_", "-"))
+					}
+				}
+				assert.ElementsMatch(t, expectedEntitlements, receivedEntitlements)
+				return nil
+			},
+		}
+
+	}
+}
+
+func makeEntitlementsSteps(t *testing.T, entitlementsSteps [][]entitlement) []step {
+	groupName := RandomName("entitlements-")
+	var c *httpclient.ApiClient
+	makeEntitlementsStep := entitlementsStepBuilder(t, &c, groupName)
+	steps := make([]step, len(entitlementsSteps))
+	for i, entitlements := range entitlementsSteps {
+		steps[i] = makeEntitlementsStep(entitlements)
+	}
+	steps[0].PreConfig = makePreconfig(t, &c, groupName)
+	return steps
+}
+
+func makePreconfig(t *testing.T, c **httpclient.ApiClient, groupName string) func() {
+	return func() {
+		w := databricks.Must(databricks.NewWorkspaceClient())
```

Comment:
would it make sense to initialize a WorkspaceClient in our CRUD methods, to catch if user is misconfiguring`databricks_entitlement` with an account-level provider?

Reply:
Yes, we just have to be careful about where we do this validation (as in, it needs to happen during the apply phase and not plan).